src/storage-types/src/connections/aws.rs

```diff
@@ -94,34 +174,45 @@ impl RustType<ProtoAwsCredentials> for AwsCredentials {
     }
 }
 
-/// A role for Materialize to assume when performing AWS API calls.
+/// Details required for Materialize to assume customer's role
 #[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash)]
 pub struct AwsAssumeRole {
     /// The Amazon Resource Name of the role to assume.
     pub arn: String,
+    /// The optional session name for the assume role session.
+    pub session_name: Option<String>,
+    /// The Materialize AWS principal details.
+    pub mz_principal: MaterializePrincipal,
 }
 
 impl RustType<ProtoAwsAssumeRole> for AwsAssumeRole {
     fn into_proto(&self) -> ProtoAwsAssumeRole {
         ProtoAwsAssumeRole {
             arn: self.arn.clone(),
+            session_name: self.session_name.clone(),
+            mz_principal: Some(self.mz_principal.into_proto()),
         }
     }
 
     fn from_proto(proto: ProtoAwsAssumeRole) -> Result<Self, TryFromProtoError> {
-        Ok(AwsAssumeRole { arn: proto.arn })
+        Ok(AwsAssumeRole {
+            arn: proto.arn,
+            session_name: proto.session_name,
+            mz_principal: proto
+                .mz_principal
+                .into_rust_if_some("ProtoAwsAssumeRole::mz_principal")?,
+        })
     }
 }
 
 impl AwsConfig {
     /// Loads the AWS SDK configuration object from the environment, then
     /// applies the overrides from this object.
-    pub async fn load(
+    pub async fn get_sdk_config(
         &self,
-        external_id_prefix: Option<&AwsExternalIdPrefix>,
-        external_id_suffix: Option<&GlobalId>,
         secrets_reader: &dyn SecretsReader,
-    ) -> aws_types::SdkConfig {
+        external_id_suffix: Option<String>,
```

Comment:
As above, recommend being explicit here that this is `connection_id: GlobalId`.

Also, the fact that this parameter is optional is a recipe for misuse. Recommend the following:

  * Split the implementation of this method to new methods `AwsCredentials::load_sdk_config` and `AwsAssumeRole::load_sdk_config`.
  * Teach this method to delegate to the new methods.
  * Replace `external_id_suffix: Option<String>` with `connection_id: GlobalId` to force all callers to do the good thing.
  * Add a `AwsAssumeRole::dangerously_load_sdk_config_without_external_id` function that shares code with `AwsAssumeRole` via a helper method or somesuch. This method can be used by `validate_connection` to test the case where the external ID is not set, but its unwieldy and scary name should prevent it from being inappropriately used in other contexts.

Reply:
Updated. 
* I added implementation `AwsCredentials::load_shared_credentials_provider` and `AwsAssumeRole::load_shared_credentials_provider` instead. Because to load the sdk_config I will also need region and endpoint which are part of the parent `AwsConfig`.
* The public interface is still `AwsConfig::load_sdk_config(&self, connection_context: &ConnectionContext, connection_id: &GlobalId)`.
* Added a `AwsAssumeRole::dangerously_load_shared_credentials_provider_without_external_id` only called by the validate method